### DISCUSSION TITLE
can an assembly be too big?

### DISCUSSION THREAD

**Discussion by szechp** [GMOD/jbrowse-components#2276] [Q&A] [answered]
Hi, i loaded a fasta file via the cli and after some time it said it was succesful. but now when i wanto to add a linear genome view in the web interface it shows me a circling loading thingy, but nothing happens. is this supposed to load that long?
![image](https://user-images.githubusercontent.com/45265588/131554589-20822e80-ac8b-43f2-aa18-ef95e45cd8ac.png)

**Comment by cmdcolin** [comment, marked as answer]
can you check what version you are using? the 1.3.1 release of jbrowse 2 included a fix (#2060) that improved performance for assemblies that have many contigs

for example, loading even a Pine tree genome with hundreds of thousands of scaffolds is pretty quick ~3 seconds after downloading the 60mb file

**Reply by szechp** [reply]
i just downloaded it 20 mins ago, so it must be the latest, let me check

**Reply by szechp** [reply]
its 1.3.5

**Reply by szechp** [reply]
the fasta file is 19gig, maybe its too much?

**Reply by cmdcolin** [reply]
the size of the fasta file shouldnt matter to much. the 60mb file I referred to is the fasta index file. in my example, e.g. the Pine tree genome I tried this

jbrowse add-assembly https://s3.amazonaws.com/jbrowse.org/genomes/Pinus_taeda/fasta/GCA_000404065.3_Ptaeda2.0_genomic.fa.gz

This fasta file is 6.5GBytes gzipped, and is measured at about 21.6Gbases

In order to load the Pine genome, the client side has to download a FASTA index file (.fai) to start with https://s3.amazonaws.com/jbrowse.org/genomes/Pinus_taeda/fasta/GCA_000404065.3_Ptaeda2.0_genomic.fa.gz.fai which is 60MB so that will be downloading while that circular thing is spinning, but after that, it should be fast.

Do you know how big your FAI file is?

**Reply by szechp** [reply]
yes i created it with samtools and its 1,35gb

**Reply by cmdcolin** [reply]
We could try to make a better download indicator for this e.g. a progress bar so that the user is not left uncertain while that FAI file is downloading...

**Reply by szechp** [reply]
im running it locally, so there should be no downloading necesary

**Reply by cmdcolin** [reply]
You are sure you have a 1.35 Gb .fai file? This might be overwhelming for the browser. There might be something we could try to do to improve performance somewhat but I think you might generally want to scaffold your genome assembly more or filter out small contigs perhaps

**Reply by szechp** [reply]
okay ill try that

**Reply by cmdcolin** [reply]
The Pine genome .fai is about 60Mb for comparison and even that is quite large, having approx 1.7M contigs/scaffolds

**Reply by szechp** [reply]
https://www.g3journal.org/content/9/7/2039 thats the genome i want to visualize, and they state that its very fragmented, so thats probably it

**Reply by cmdcolin** [reply]
interesting... it looks at least about 10x more fragmented than the p.taeda which was our previous benchmark for loading large fragmented assemblies. N50 of 14kb vs >200kb ...there might be some way we could help as it may not really be strictly necessary to load the fai file on startup. let us know if you think that might be useful

**Reply by szechp** [reply]
that sounds good, but maybe the fragmented nature of the genome is the main problem? i guess the fai has to be loaded at some point, so the problemn that it is too big might not be solved? and unrelated because im a real beginner in this field, what would you recommend to "simplify" the fai file? to i have to reassembly it or can i edit the fasta somehow?

**Reply by cmdcolin** [reply]
It is hard to say. You could potentially filter small contigs like this https://www.biostars.org/p/129796/ 

another option would be to arbitrarily join some of the contigs, separated by gaps of Ns, into fake scaffolds. this would not be lossy operation like filtering it out, but would potentially be a little confusing

 the ideal thing would be for a group to make a new genome assembly incorporating long reads or hi-c to assist with joining the fragments together

**Reply by szechp** [reply]
Okay that biostars link did the trick, the fai is now ~50mb and loads just fine.  Thanks for your help!
